### PR TITLE
.NET Core

### PR DESCRIPTION
**Pull request by droyad**

### CODE REVIEW

File: source/Octostache.Tests/UsageFixture.cs [hunk 1]
```diff
@@ -91,10 +91,9 @@ public void EmptyValuesAreEchoed(string template, string variableDefinitions, st
         [Test]
         [TestCase("#{Foo}", "Foo=#{Foo}")]
         [TestCase("#{Foo}", "Foo=#{Fox};Fox=#{Fax};Fax=#{Fix};Fix=#{Foo}")]
-        [ExpectedException(typeof(InvalidOperationException))]
         public void MaximumRecursionLimitException(string template, string variableDefinitions)
         {
-            ParseVariables(variableDefinitions).Evaluate(template);
+            Assert.That(() => ParseVariables(variableDefinitions).Evaluate(template), Throws.InvalidOperationException);
```
Comment (droyad) [inline source/Octostache.Tests/UsageFixture.cs]: Would bringing in FluentAssertions make this clearer?

File: source/Octostache/Templates/BuiltInFunctions.cs [hunk 2]
```diff
@@ -67,8 +67,9 @@ static string Markdown(string raw)
             var options = new MarkdownOptions();
             options.AutoHyperlink = true;
             options.LinkEmails = true;
+            options.AllowEmptyLinkText = true;
 
-            return new Markdown(options).Transform(raw.Trim());
+            return new Markdown(options).Transform(raw.Trim()) + "\n";
```
Comment (droyad) [inline source/Octostache/Templates/BuiltInFunctions.cs]: No `\r`?

File: source/Octostache/Templates/TemplateParser.cs [hunk 3]
```diff
@@ -216,23 +220,53 @@ static Parser<string> Keyword(string text)
...
         {
             Cache = new MemoryCache("Octostache", new NameValueCollection() { { "CacheMemoryLimitMegabytes", (20 * 1024).ToString() } });
         }
 
+        private static void AddToCache(string template, Template cached)
+        {
+            Cache.Set(template, cached, new CacheItemPolicy() { SlidingExpiration = TimeSpan.FromMinutes(10) });
+        }
+        
+        private static Template GetFromCache(string template)
```
Comment (droyad) [inline source/Octostache/Templates/TemplateParser.cs]: This can go outside the #if

File: source/Octostache/project.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  "frameworks": {
+    "net451": {
+      "frameworkAssemblies": {
+        "System.Core": "*",
+        "System.Runtime.Caching": "*",
+        "Microsoft.CSharp": "*"
+      },
+      "buildOptions": {
+        "define": [ "NET46" ]
```
Comment (droyad) [inline source/Octostache/project.json]: Should we standardise on NET40 (defining the runtime, not the framework)? It also doesn't match the actual framework in this case.

File: source/Octostache/Templates/TemplateParser.cs [hunk 5]
```diff
@@ -3,7 +3,11 @@
 using System.Collections.Specialized;
 using System.Globalization;
 using System.Linq;
+#if NET46
 using System.Runtime.Caching;
+#else
+using Microsoft.Extensions.Caching.Memory;
```
Comment (droyad) [inline source/Octostache/Templates/TemplateParser.cs]: Should we use this library in all cases, keeping functionality consistent between target frameworks?
Perhaps we need to do our own cache size limiting.

File: source/Octostache.Tests/UsageFixture.cs (hunk 1 above)
Comment (matt-richardson) [inline source/Octostache.Tests/UsageFixture.cs]: Not convinced - bringing in a full library for just a single assertion doesn't sound great.

File: source/Octostache/Templates/BuiltInFunctions.cs (hunk 2 above)
Comment (matt-richardson) [inline source/Octostache/Templates/BuiltInFunctions.cs]: Nope - its a difference between `MarkdownSharp` and `Heyred.MarkdownSharp`.
Specifically:
screen shot 2016-08-10 at 10 24 52
`MarkdownSharp` appends a "\n", whereas `HeyRed.MarkdownSharp` doesn't.

I'm tempted to pull in the unit tests for `MarkdownSharp` to ensure we get like for like behaviour with existing.

File: source/Octostache/Templates/TemplateParser.cs (hunk 5 above)
Comment (matt-richardson) [inline source/Octostache/Templates/TemplateParser.cs]: I considered that... Ended up on the fence. Worth discussing more.
I'll investigate the cache size limiting.

**Comment by matt-richardson** [comment]
Closing PR while I'm getting AppVeyor build to work

**Pull request closed without merging**